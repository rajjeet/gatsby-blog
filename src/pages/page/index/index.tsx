import React from 'react';
import { graphql } from 'gatsby';
import Layout from '../../../components/layout';
import PostListing from '../../../components/post-listing';
import Seo from '../../../components/seo';
import ProjectListing from '../../../components/project-listing';
import { TPostEdge, TImage } from '../../../templates/category-page/types';

export type TProjectEdge = {
  node: {
    frontmatter: {
      title: string;
      description: string;
      thumbnail: TImage;
      techStackTags: {label: string; type: string}[];
      links: {label: string; value: string}[];
    };
    fields: {
      slug: string;
    };
    body: string;
  };
};

export type TProps = {
  data: {
    posts: {
      edges: TPostEdge[];
    };
    projects: {
      edges: TProjectEdge[];
    };
  };
};

const IndexPage: React.FC<TProps> = ({ data }) => (
  <Layout>
    <Seo />
    <ProjectListing
      projects={data.projects.edges}
      heading="Side Projects"
    />
    <PostListing
      posts={data.posts.edges}
      heading="Latest Posts"
    />
  </Layout>
);

export default IndexPage;

export const query = graphql`
    {
        posts: allMdx(sort: {fields: [frontmatter___date], order: DESC}, filter: {frontmatter: {draft: {ne: true}}, fields: {contentType: {eq: "post"}}}, limit: 3) {
            edges {
                node {
                    ...PostListingMarkdownFragment
                }
            }
        }
        projects: allMdx(limit: 2, sort: {fields: frontmatter___date, order: DESC}, filter: {fields: {contentType: {eq: "project"}}}) {
            edges {
                node {
                    ...ProjectListingMarkdownFragment
                }
            }
        }
    }
`;
