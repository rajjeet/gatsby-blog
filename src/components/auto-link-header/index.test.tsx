import React from 'react';
import { render } from '@testing-library/react';
import { AutoLinkHeader } from './index';

describe('<AuthorCard />', () => {
  const Component = AutoLinkHeader('h1');
  it('should render', () => {
    const { asFragment } = render(<Component>Sample Heading</Component>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should show the heading text', () => {
    const { getByText } = render(<Component>Sample Heading</Component>);
    expect(getByText('Sample Heading')).toBeDefined();
  });
});

