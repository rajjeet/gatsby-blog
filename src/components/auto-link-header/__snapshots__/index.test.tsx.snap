// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorCard /> should render 1`] = `
<DocumentFragment>
  <h1
    id="sample-heading"
  >
    <a
      aria-hidden="true"
      href="sample-heading"
    />
    Sample Heading
  </h1>
</DocumentFragment>
`;
